aiosmtpd/tests/test_main.py

```diff
@@ -92,43 +92,59 @@ def setuid(mocker):
 # region ##### Helper Funcs ###########################################################
 
 
-def watch_for_tls(ready_flag, has_tls, req_tls):
-    has_tls.value = False
-    req_tls.value = False
+def watch_for_tls(ready_flag, retq: MP.Queue):
+    has_tls = False
+    req_tls = False
     ready_flag.set()
     start = time.monotonic()
-    while (time.monotonic() - start) <= AUTOSTOP_DELAY:
+    delay = AUTOSTOP_DELAY * 1.5
+    while (time.monotonic() - start) <= delay:
         try:
             with SMTPClient("localhost", 8025) as client:
                 resp = client.docmd("HELP", "HELO")
                 if resp == S.S530_STARTTLS_FIRST:
-                    req_tls.value = True
+                    req_tls = True
                 client.ehlo("exemple.org")
                 if "starttls" in client.esmtp_features:
-                    has_tls.value = True
-                return
+                    has_tls = True
+                break
         except Exception:
             time.sleep(0.05)
+    retq.put(has_tls)
+    retq.put(req_tls)
 
 
-def watch_for_smtps(ready_flag, has_smtps):
-    has_smtps.value = False
+def watch_for_smtps(ready_flag, retq: MP.Queue):
+    has_smtps = False
     ready_flag.set()
     start = time.monotonic()
-    while (time.monotonic() - start) <= AUTOSTOP_DELAY:
+    delay = AUTOSTOP_DELAY * 1.5
+    while (time.monotonic() - start) <= delay:
         try:
             with SMTP_SSL("localhost", 8025) as client:
                 client.ehlo("exemple.org")
-                has_smtps.value = True
-                return
+                has_smtps = True
+                break
         except Exception:
             time.sleep(0.05)
+    retq.put(has_smtps)
 
 
 def main_n(*args):
     main(("-n",) + args)
 
 
+@contextmanager
+def watcher_process(func):
+    redy = MP.Event()
+    retq = MP.Queue()
+    proc = MP.Process(target=func, args=(redy, retq))
+    proc.start()
+    redy.wait()
+    yield retq
+    proc.join()
```

Comment:
Boilerplate code to (1) start a child process, (2) wait until child process is ready, and (3) join child process when done.